playbooks/new-tenant-juju/pre.yaml

```diff
@@ -22,9 +22,12 @@
         # ospurge requires cryptography, and we don't want to have to bring
         # in the setuptools_rust toolchain as well for this, it's not needed
         - 'cryptography<3.4'
+    - name: 'prepare tenant name'
+      set_fact:
+        tenant_project_name_raw: "zosci-test-{{ zuul.change }}-{{ zuul.project.short_name }}-{{ zuul.build }}"
     - name: 'create tenant name'
       set_fact:
-        tenant_project_name: "zosci-test-{{ zuul.job }}-{{ zuul.project.short_name }}-{{ zuul.build }}"
+        tenant_project_name: "{{ tenant_project_name_raw | truncate(64, True, '') }}"
```

Comment:
Is it guaranteed that after truncation, the name will still be unique? Is `zuul.change` unique already?

Reply:
change is, I think, the 6 digit number on the change in gerrit, the zuul.build on the end is a uuid for this specific build, I'm truncating off that part so it _could_ potentially clash if the project's name is long (I'm looking at you charm-neutron-api-plugin-ovn) but that should still be fairly uncommon

Reply:
I'm actually tempted to remove the `short_name` as we can get there easily enough from the change

Reply:
yes or move the short_name to the end, so truncation can't impact uniqueness